### PR TITLE
chore: reduce more epoch log noise

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

This PR aims to reduce more noise that occurs during epoch logs:

Screenshot 2024-03-04 at 3 53 35 PM

Screenshot 2024-03-04 at 3 53 14 PM

Errors occur when attempting to swap fees collected to a specific denom, usually due to a lack of liquidity in the pool that exists. Since this is expected, for this epoch loop of swapping fees, we:

1. Move the individual error logs to debug logs
2. Emit an info log with the aggregated data (how much we attempted to swap across what pools) that did not get swapped this epoch, and if specifics are desired, the debug logs can be reviewed.

## Testing and Verifying

I ran this logic on v23 mainnet during the last epoch and it got through with no issues, while only emitting the desired logs.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

## Summary by CodeRabbit

- **New Features**
	- Enhanced logging capabilities for certain operations to aid in debugging.
- **Improvements**
	- Improved tracking and logging of token swap errors to enhance transparency and troubleshooting.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -28,7 +28,7 @@ require (
 	github.com/ory/dockertest/v3 v3.10.0
 	github.com/osmosis-labs/go-mutesting v0.0.0-20221208041716-b43bcd97b3b3
 	github.com/osmosis-labs/osmosis/osmomath v0.0.9-0.20240222171503-685566578734
-	github.com/osmosis-labs/osmosis/osmoutils v0.0.9-0.20240222004208-b602d1901059
+	github.com/osmosis-labs/osmosis/osmoutils v0.0.9-0.20240304224617-9042f6d1e24f
```
Comment (mattverse) [inline go.mod:31]: Can we also change other osmoutil instances from different `go.mod` in our repo to this commit? (really should not matter, but kind of feels weird having different commits of osmoutils being imported)

**Review by mattverse — APPROVED** [review]
LGTM, wonder if there's anyway we can remove code duplication with `ApplyFuncIfNoError`

**Comment by czarcas7ic** [comment]
@mattverse I reduced the code duplication and ran the action to update all go mod imports

**Comment by mattverse** [comment]
@czarcas7ic  ace! LGTM 😎

File: x/txfees/keeper/hooks.go [hunk 2]
```diff
@@ -224,6 +224,8 @@ func (k Keeper) calculateDistributeAndTrackTakerFees(ctx sdk.Context, defaultFee
 func (k Keeper) swapNonNativeFeeToDenom(ctx sdk.Context, denomToSwapTo string, feeCollectorAddress sdk.AccAddress) sdk.Coin {
 	coinsToSwap := k.bankKeeper.GetAllBalances(ctx, feeCollectorAddress)
 	totalCoinOut := sdk.NewCoin(denomToSwapTo, osmomath.ZeroInt())
+	coinsNotSwapped := []string{}
```
Comment (ValarDragon) [inline x/txfees/keeper/hooks.go:227]: We don't do anything with this variable? Presumably were supposed to log it if its not empty?

Comment (czarcas7ic) [reply x/txfees/keeper/hooks.go:227]: I think it got removed when I merged in main :/

Comment (czarcas7ic) [reply x/txfees/keeper/hooks.go:227]: added back https://github.com/osmosis-labs/osmosis/pull/7660/commits/efaa89efe751206a4f059f2dc268c79775da7f68

**Review by ValarDragon — APPROVED** [review]

**Pull request merged**